modules/dnn/test/test_layers.cpp

```diff
@@ -690,22 +697,27 @@ TEST_P(Scale_untrainable, Accuracy)
     float* refData = (float*)ref.data;
     float* weightsData = (float*)weights.data;
     int spatialSize = 1;
-    for (int i = axis + weightsDims; i < 4; ++i)
+    for (int i = axis + weightsDims - 1; i < 4; ++i)
         spatialSize *= inpShape[i];
-    for (int i = 0; i < ref.total(); ++i)
+    //CV_Error(Error::StsNotImplemented, std::to_string(spatialSize));
+    for (int num_samples = 0; num_samples < inpShape[0] ;num_samples++)
     {
-        float w = weightsData[(i / spatialSize) % weights.total()];
-        refData[i] = inpData[i] * w;
+        for (int i = 0; i < totalPerSample; ++i)
+        {
+            int index = i + num_samples * totalPerSample;
+            float w = weightsData[(i / spatialSize) % totalWeightsPerSample + num_samples * totalWeightsPerSample ];
+            refData[index] = inpData[index] * w;
+        }
     }
     normAssert(out, ref);
 }
 
 INSTANTIATE_TEST_CASE_P(Layer_Test, Scale_untrainable, Combine(
 /*input size*/   Values(Vec4i(2, 3, 4, 5)),
 /*axis, #dims*/  Values(Vec2i(0, 1), Vec2i(0, 2), Vec2i(0, 3), Vec2i(0, 4),
-                                     Vec2i(1, 1), Vec2i(1, 2), Vec2i(1, 3),
-                                                  Vec2i(2, 1), Vec2i(2, 2),
-                                                               Vec2i(3, 1)),
+                                     Vec2i(1, 2), Vec2i(1, 3), Vec2i(1, 4),
+                                     Vec2i(2, 2), Vec2i(2, 3),
+                                     Vec2i(3, 2)),
```

Comment:
Sorry, couldn't understand.

> And just add one more parameter for batch size.

by this you mean `axis, dims, batchsize` ? 
Batchsize is always 2 in these tests and original parameters are still there. `axis, dims`.
I just changed dims by adding 1 because previously batchsize wasn't considered in dims.

Reply:
So why do we need to modify this test? Can you please clarify a bit

Reply:
```cpp
/*axis, #dims*/  Values(Vec2i(0, 1), Vec2i(0, 2), Vec2i(0, 3), Vec2i(0, 4),
                                     Vec2i(1, 1), Vec2i(1, 2), Vec2i(1, 3),
                                                  Vec2i(2, 1), Vec2i(2, 2),
                                                               Vec2i(3, 1)),
```

shape of input [ N, C, H, W ] 

`Vec2i(0, 1) - axis = 0, dim = 1 [ N ]` fine
`Vec2i(0, 2) - axis = 0, dim = 2 [ N, C ]` fine
`Vec2i(0, 3) - axis = 0, dim = 2 [ N, C, H ]` fine
`Vec2i(0, 4) - axis = 0, dim = 2 [ N, C, H, W ]` fine
`Vec2i(1, 1) - axis = 1, dim = 1 [ N ]` , seems incorrect 
`Vec2i(1, 2) - axis = 1, dim = 2 [ N, C ]` - fine
`Vec2i(1, 3) - axis = 1, dim = 3 [ N, C, H ]` - fine
`Vec2i(2, 1) - axis = 2, dim = 1 [ N ]`, seems incorrect 
`Vec2i(2, 2) - axis = 2, dim = 2 [ N, H ]`- fine, can be broadcasted as  [ N, 1, H ] 
`Vec2i(3, 1) - axis = 3, dim = 1 [ N ]`- seems incorrect

So I removed these three tests [ (1, 1), (2, 1), (3, 1)]. and added three more test that are possible [ (1, 4), (2, 3), (3, 2) ]:
`Vec2i(1, 4) - axis = 1, dim = 4 [ N, C, H, W ]`- fine
`Vec2i(2, 3) - axis = 2, dim = 3 [ N, C, H ]`- fine
`Vec2i(3, 2) - axis = 3, dim = 2 [ N, W ]`- fine, can be broadcasted as [ N, 1, 1, W ]

```cpp

/*axis, #dims*/  Values(Vec2i(0, 1), Vec2i(0, 2), Vec2i(0, 3), Vec2i(0, 4),
                                     Vec2i(1, 2), Vec2i(1, 3), Vec2i(1, 4),
                                     Vec2i(2, 2), Vec2i(2, 3),
                                     Vec2i(3, 2)),
```

Reply:
Can you please add a separate test only for new functionality? If I'm not mistaken, we try so solve batching in Scale Layer. So while
```
1 x C x H x W
1 x C
```
work can we test
```
N x H x W x C
1 x C
```
and
```
N x C x H x W
N x C
```
?

Reply:
sure. they work. Let me add tests for them.

Reply:
https://github.com/opencv/opencv/blob/2842116ebd717356ecc8f222284161ae5f0cb4a9/modules/dnn/test/test_layers.cpp#L634
these tests will cover 
```
N x C x H x W
N x C
```
https://github.com/opencv/opencv/blob/2842116ebd717356ecc8f222284161ae5f0cb4a9/modules/dnn/test/test_layers.cpp#L723
and these will cover 
```
N x H x W x C
1 x C
```

Reply:
Please keep existing test unchanged.

Reply:
@dkurt Done!

Reply:
I didn't remove any but added some more tests.